src/pages/iv-items/forms/SaleList.js

```diff
@@ -0,0 +1,289 @@
+import { useState, useContext, useEffect } from 'react'
+import { useWindow } from 'src/windows'
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import * as yup from 'yup'
+
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import { useForm } from 'src/hooks/form.js'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { Grid } from '@mui/material'
+import SalesForm from './SalesForm'
+import FormShell from 'src/components/Shared/FormShell'
+
+const SalesList = ({ store, labels, maxAccess, formikInitial }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { recordId } = store
+  const { _msId } = store
+  const { measurementId } = store
+  const { priceGroupId } = store
+  const { returnPolicy } = store
+
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  const [check, setCheck] = useState(false)
+  const [firstCurr, setFirstCurr] = useState(false)
+  useEffect(() => {
+    if (recordId) {
+      const fetchCurrency = async () => {
+        try {
+          const response = await getRequest({
+            extension: InventoryRepository.Currency.qry,
+            parameters: `&_itemId=${recordId}`
+          })
+          if (response.list && response.list.length > 0) {
+            setFirstCurr(response.list[0].currencyId)
```

Comment:
I guess there's no need for setFirstCurr useState also, u can here update currency value in formik directly `formik.setFieldValue('currencyId',response.list[0].currencyId)`